src/main/logutils.ts

```diff
@@ -373,32 +376,132 @@ const determineArenaMatchResult = (line: LogLine): any[] => {
 }
 
 /**
- * Guard functions for now to simply allow us to ignore
- * Mythic Keystone encounters because we can't actually handle them
- * right now.
+ * Handle a log line for CHALLENGE_MODE_START
  */
-function handleChallengeModeStartLine (_line: LogLine): void {
-    isChallengeModeActive = true;
-}
+function handleChallengeModeStartLine (line: LogLine): void {
+    // It's impossible to start a keystone dungeon while another one is in progress
+    // so we'll just remove the existing one and make a new one when `CHALLENGE_MODE_START`
+    // is encountered.
+    if (activeChallengeMode) {
+        console.warn("[ChallengeMode] A Challenge Mode instance is already in progress; abandoning it.")
+    }
+    videoStartDate = line.date();
 
-function handleChallengeModeEndLine (_line: LogLine): void {
-    isChallengeModeActive = false;
-}
+    const zoneName = line.args[2];
+    const mapId = parseInt(line.args[3], 10);
+
+    if (!(mapId in dungeonTimersByMapId && mapId in dungeonsByMapId)) {
```

Comment:
> This is a bit ugly, can we split it out?

I'm generally against making variables that are only used in a single following comparison, but if that's what you want, I'll do it.

> Was also suprised to see no exit from this function here, don't you need to return to avoid processing a dungeon we don't recognise?

Yes and no. I was hoping to and planning on making it so that in the future when new dungeons are added, we wouldn't avoid recording and saving videos from unknown dungeons, encounters and such so that we could simply update the app and the details would show up after the recording had been saved.

`handleEncounterStartLine()` also doesn't care about the encounter existing or not.

Reply:
Ah, I am of the mindset that if it makes code readable then it's a good thing. I appreciate that's an opinion not a fact so I'm happy either way.

Reply:
I split it. It's your project afterall :)